src/Airy/cairy.jl

```diff
@@ -377,197 +377,98 @@ end
     else
-        out = 2*d + im*c
-        check_conj && (out = conj(out))
-        return out
+        ai = A
+        aip = C
     end
-end
-
-# see equations 24 and relations using eq 25 and 26 in [1]
-function airy_large_arg_a(x::ComplexOrReal{T}; tol=eps(T)*40) where T
-    S = eltype(x)
-    MaxIter = 3000
-    xsqr = sqrt(x)
 
-    out = zero(S)
-    t = GAMMA_ONE_SIXTH(T) * GAMMA_FIVE_SIXTHS(T) / 4
-    a = 4*xsqr*x
-    for i in 0:MaxIter
-        out += t
-        abs(t) < tol*abs(out) && break
-        t *= -3*(i + one(T)/6) * (i + T(5)/6) / (a*(i + one(T)))
-    end
-    return out * exp(-a / 6) / (sqrt(T(π)^3) * sqrt(xsqr))
+    xsqr = sqrt(xsqr)
+    return ai * Base.FastMath.inv_fast(xsqr) * inv(PIPOW3O2(T)), aip * xsqr * inv(PIPOW3O2(T))
 end
 
-function airy_large_arg_b(x::ComplexOrReal{T}; tol=eps(T)*40) where T
-    S = eltype(x)
-    MaxIter = 3000
-    xsqr = sqrt(x)
-
-    out = zero(S)
-    t = GAMMA_ONE_SIXTH(T) * GAMMA_FIVE_SIXTHS(T) / 4
-    a = 4*xsqr*x
-    for i in 0:MaxIter
-        out += t
-        abs(t) < tol*abs(out) && break
-        t *= 3*(i + one(T)/6) * (i + T(5)/6) / (a*(i + one(T)))
+# valid in 0 <= angle(z) <= pi
+# use conjugation for bottom half plane
+@inline function airybix_large_args(z::Complex{T}) where T
+    xsqr = sqrt(z)
+    xsqrx = Base.FastMath.inv_fast(z * xsqr)
+    A, B, C, D = compute_airy_asy_coef(z, xsqrx)
+    
+    if abs(angle(z)) < 2π/3
+        B *= 2
+        D *= 2
     end
-    return out * exp(a / 6) / (sqrt(T(π)^3) * sqrt(xsqr))
-end
 
-function airy_large_arg_c(x::ComplexOrReal{T}; tol=eps(T)*40) where T
-    S = eltype(x)
-    MaxIter = 3000
-    xsqr = sqrt(x)
+    e = exp(-4/3 * z * xsqr)
+    xsqr = sqrt(xsqr)
 
-    out = zero(S)
-    # use identities of gamma to relate to defined constants
-    # t = gamma(-one(T) / 6) * gamma(T(7) / 6) / 4
-    t = -GAMMA_FIVE_SIXTHS(T) * GAMMA_ONE_SIXTH(T) / 4
-    a = 4*xsqr*x
-    for i in 0:MaxIter
-        out += t
-        abs(t) < tol*abs(out) && break
-        t *= -3*(i - one(T)/6) * (i + T(7)/6) / (a*(i + one(T)))
-    end
-    return out * exp(-a / 6) * sqrt(xsqr) / sqrt(T(π)^3)
+    bi = muladd(A*im, e, B) * Base.FastMath.inv_fast(xsqr) * inv(PIPOW3O2(T))
+    bip = muladd(C*im, e, -D) * xsqr * inv(PIPOW3O2(T))
+    return bi, bip
 end
 
-function airy_large_arg_d(x::ComplexOrReal{T}; tol=eps(T)*40) where T
-    S = eltype(x)
-    MaxIter = 3000
-    xsqr = sqrt(x)
-
-    out = zero(S)
-    # use identities of gamma to relate to defined constants
-    # t = gamma(-one(T) / 6) * gamma(T(7) / 6) / 4
-    t = -GAMMA_FIVE_SIXTHS(T) * GAMMA_ONE_SIXTH(T) / 4
-    a = 4*xsqr*x
-    for i in 0:MaxIter
-        out += t
-        abs(t) < tol*abs(out) && break
-        t *= 3*(i - one(T)/6) * (i + T(7)/6) / (a*(i + one(T)))
-    end
-    return -out * exp(a / 6) * sqrt(xsqr) / sqrt(T(π)^3)
+@inline function compute_airy_asy_coef(z, xsqrx)
+    invx3 = @fastmath inv(z^3)
+    p1 = evalpoly(invx3, AIRY_ASYM_COEF[1])
```

Comment:
SIMDMath here?

Reply:
Ya might as well make the dependency now !! (I set this up so it will be easy switch). I think the complex SIMD_evalpoly method is working now.

Reply:
See comment below. Need to tag another version of SIMDMath to better handle negated muladd for complex numbers